clusterman/util.py

```diff
@@ -418,3 +424,25 @@ def strtobool(val: str) -> int:
         return 0
     else:
         raise ValueError("invalid truth value %r" % (val,))
+
+
+def limit_function_runtime(
+    func: Callable[[], Any], max_runtime: Union[int, float], callback: Optional[Callable[[], None]] = None
```

Comment:
super-nit: it might be nice to use [ParamSpec/TypeVar](https://docs.python.org/3/library/typing.html#typing.ParamSpec) to type `func` (actually, maybe we just need TypeVar since we only expect 0-arg functions to be passed)

Reply:
I don't really see a use for that though, since the parameter type is not re-referenced in any other place.